### PR TITLE
feat - Additions needed for integrating to pmedianetwork/adverity

### PR DESCRIPTION
**Pull request by bebraw**
This is the first chunk of additions/changes required by https://github.com/pmedianetwork/adverity/issues/11398.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -82,12 +82,12 @@ function loadJavaScript({
 } = {}): webpack.Configuration {
   return {
     resolve: {
-      extensions: [".js"],
+      extensions: [".jsx", ".js"],
```
Comment (TkDodo) [inline src/index.ts:85]: I don't think that we are using the `jsx` extension anywhere - not even in `insights`. I see that our current webpack config in `insights` has it, but I think it is a leftover from earlier times. We moved away from the `.jsx` extension and use `.js` only. We could keep it though, as it doesn't do any harm.

**Review by TkDodo — APPROVED** [review]
Will this have any impacts on `datatap` when we use this config there as well?

File: src/index.ts (hunk 1 above)
Comment (bebraw) [reply src/index.ts:85]: Ah, I see. Good to know!

Let's keep it then.

**Comment by bebraw** [comment]
> Will this have any impacts on datatap when we use this config there as well?

For small image/font assets, it's going to apply the inlining configuration from adverity/adverity so for any asset that's sufficiently small, you'll see difference in output. The other changes are non-breaking and I took care to do so in the function interfaces.

That said, we likely should test easily breakable bits like regexps and add more rigidity to the repository so it's harder to break. In terms of integration testing to make sure pushing a new release won't break things, likely this repository could have a mini-project using the different parts as then you can assert against the project output (think snapshots) but we can discuss this later.

I'll merge this now and continue on the bits at the main PR (https://github.com/pmedianetwork/adverity/pull/11404).

**Comment by TkDodo** [comment]
> For small image/font assets, it's going to apply the inlining configuration from adverity/adverity so for any asset that's sufficiently small, you'll see difference in output. The other changes are non-breaking and I took care to do so in the function interfaces.

This seems like a good change 👍 

> I'll merge this now and continue on the bits at the main PR (pmedianetwork/adverity#11404).

👍

**Pull request merged**